jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/jpa/PersistenceProvider.java

```diff
@@ -316,29 +320,59 @@ public JPAInitializer getInitializer(String emName, Map m){
         return JavaSECMPInitializer.getJavaSECMPInitializer(classLoader);
     }
 
+    // Package visibility is required for jUnit
     /**
-     * Need to check that the provider property is null or set for EclipseLink
+     * The Persistence bootstrap class must locate all the persistence providers using the PersistenceProviderResolver
+     * mechanism described in Section 9.3 and call createEntityManagerFactory on them in turn until an appropriate backing
+     * provider returns an EntityManagerFactory instance. A provider may deem itself as appropriate for the persistence unit
+     * if any of the following are true:
+     * <p> * Its implementation class has been specified in the provider element for that persistence unit in the persistence.xml
+     * file and has not been overridden by a different jakarta.persistence.provider property value included in the Map passed
+     * to the createEntityManagerFactory method.
+     * <p> * The jakarta.persistence.provider property was included in the Map passed to createEntityManagerFactory and the value
+     * of the property is the provider’s implementation class.
+     * <p> * No provider was specified for the persistence unit in either the persistence.xml or the property map.
+     *
+     * @since Jakarta Persistence 3.2
      */
-    public boolean checkForProviderProperty(Map properties){
-        Object provider = properties.get("jakarta.persistence.provider");
-        if (provider != null){
-            //user has specified a provider make sure it is us or abort.
-            if (provider instanceof Class){
-                provider = ((Class)provider).getName();
-            }
-            try{
-                if (!(EntityManagerFactoryProvider.class.getName().equals(provider) || PersistenceProvider.class.getName().equals(provider))){
-                    return false;
-                    //user has requested another provider so lets ignore this request.
-                }
-            }catch(ClassCastException e){
-                return false;
-                // not a recognized provider property value so must be another provider.
-            }
+    static boolean isProviderEclipseLink(PersistenceConfiguration configuration) {
```

Comment:
This method uses nothing from the instance so there is no need to be non static.
Also this check is quite common feature so it may be reused elsewhere when needed in the future, just visibiliti may be raised.